src/main/cli/cli.c

```diff
@@ -2703,6 +2707,58 @@ static void cliVtx(const char *cmdName, char *cmdline)
 
 #endif // VTX_CONTROL
 
+#ifdef USE_VTX_SMARTAUDIO
+static void cliVtxOpModel(const char *cmdName, char *cmdline)
+{
+    if (isEmpty(cmdline)) {
+        // If you change the model this value will be stale until reboot.
+        cliPrintLine(saDevice.willBootIntoPitMode ? "race" : "free");
+    } else {
+        bool freeModel = strncasecmp(cmdline, "free", 4) == 0;
+        bool pir = strncasecmp(cmdline, "pir", 3) == 0 || strncasecmp(cmdline, "race", 4) == 0;
+        bool por = strncasecmp(cmdline, "por", 3) == 0;
+        char *unsupported = "%s is unsupported by SA %s";
+
+        if (saDevice.version == 1) {
+            // V1 devices don't support pit mode.
+            if (!freeModel) {
+                cliPrintErrorLinef(cmdName, unsupported, cmdline, "1.0");
+            }
+            return;
+        }
+
+        int mode;
+
+        if (freeModel) {
+            // VTX should power up transmitting - unset PIR and POR bits.
+            mode = 0;
+        } else if (pir) {
+            mode = SA_MODE_SET_IN_RANGE_PITMODE;
+
+            if (saDevice.version >= 3 && !(saDevice.mode & SA_MODE_GET_PITMODE)) {
+                mode |= SA_MODE_CLR_PITMODE;
+            }
+        } else if (por) {
+            // V2.1 devices only supports PIR mode.
+            if (saDevice.version >= 3) {
+                cliPrintErrorLinef(cmdName, unsupported, cmdline, "2.1");
+                return;
+            }
+
+            mode = SA_MODE_SET_OUT_RANGE_PITMODE;
+        } else {
+            cliPrintErrorLinef(cmdName, "Invalid option");
+            return;
+        }
+
+        saSetMode(mode);
```

Comment:
It is not the right approach to drag a dependency on the SmartAudio driver into this - this should be done in a device agnostic way by using `vtxCommonSetPitMode()`.

Reply:
`vtxCommonSetPitMode` is just about turning pit mode on and off _once the VTX is powered up_. It's the logic that's called when you:
* flip the pit-mode toggle in the Configurator.
* update the pit-mode value on your transmitter using the `vtx.lua` script.

[ When you use the SmartAudio OSD menus, it directly calls the relevant SmartAudio functions and bypasses the device-agnostic functions like `vtxCommonSetPitMode`. ]

However, this PR is not doing the same thing as `vtxCommonSetPitMode` as it's not about immediately switching in and out of pit mode. It's about configuring what the VTX will do _when it is next powered up_ - will it start in pit mode (`race`) or will it start transmitting at full power (`free`). Configuring this behavior is quite SmartAudio specific and is not covered by `vtxCommon` functionality.

As it's SmartAudio specific, I've surrounded this function, the `#include` above and the `CLI_COMMAND_DEF` below with `#ifdef USE_VTX_SMARTAUDIO` so that this logic is not included unless you already have a dependency on the SmartAudio driver (as is the case for all or most of the standard targets).

[ @mikeller - thanks for your three comments and for taking the time to thoroughly read through my first Betaflight PR and provide constructive feedback, it's much appreciated. ]

Reply:
I could actually create a hybrid where I got `vtxCommonSetPitMode` to do some of the bit flipping but I'd still need to work with SA specific stuff here as I'd need to be able to set the `willBootIntoPitMode` and `mode` fields of `smartAudioDevice_s` so that `vtxCommonSetPitMode` went down the required paths.

However, such a hybrid wouldn't eliminate the need for SA-specific logic here and it would introduce an opaque dependency on the inner-workings of the SA implementation of the `vtxCommonSetPitMode` logic.

Reply:
If this functionality is available on SmartAudio devices, and there is a use case for users to want to set it, then we should look into making this settable through the VTX stack - having every place where this can potentially be set (CLI, CMS, MSP, Spektrum VTX control, ...) implement an insular solution that circumvents the VTX stack completely defeats the purpose of having a unified VTX stack in the first place.
We could potentially add the boot mode as a bit flag on the `uint8_t onOff` parameter of `vtxCommonSetPitMode()`.

Reply:
@mikeller wrote:
> If this functionality is available on SmartAudio devices, and there is a use case for users to want to set it, then...

There is a use case for this functionality and many users already can and do set it. It's already useable (and implemented much as here) _if you have an OSD_ and can use the SmartAudio specific OSD submenu as described [here](https://github.com/betaflight/betaflight/wiki/Unify-Smartaudio#config-sub-menu).

But some users, like me, don't have an OSD and so have no way of accessing this existing functionality. This PR _isn't_ about adding functionality that wasn't already available. It's simply about making an existing feature available to users who, for whatever reason, aren't in a position to access it via the OSD.

If you look at the SmartAudio specific OSD menu code in [`cms_menu_vtx_smartaudio.c`](https://github.com/betaflight/betaflight/blob/4.2.5/src/main/cms/cms_menu_vtx_smartaudio.c) you'll see that it actually does use `vtxCommonSetPitMode` [here](https://github.com/betaflight/betaflight/blob/4.2.5/src/main/cms/cms_menu_vtx_smartaudio.c#L527) when it wants to actively toggle pit mode on and off immediately. But when it wants to configure the startup behavior of the VTX it resorts to SmartAudio specific calls (in particular `saSetMode`) in `saCmsConfigOpmodelByGvar` and its companion function `saCmsConfigPitFModeByGvar`.

So it's just this existing VTX configuration capability from the SmartAudio specific OSD logic that I want to make available to non-OSD users.

I do understand what you're saying about expanding the existing unified VTX stack. I could:

*  Expand `vtxCommonGetStatus` in order to not just retrieve whether pit mode is currently on or off but also retrieve how the VTX is configured in relation to pit mode.
* Expand `vtxCommonSetPitMode` so that it not only switches pit mode on or off but also manages the VTX's pit mode related configuration.

However, I would say that the existing `cms_menu_vtx_smartaudio.c` logic hasn't been held to this standard (it uses SA specific logic for managing configuration) but, _more importantly_, I think managing configuration in a stack agnostic fashion will always be a bit hacky - immediate on/off is simple enough behavior, suitable for `vtxCommon` control, but configuration is always probably going to be a bit stack specific. E.g. SmartAudio has this idea of in-range and out-range pit mode.

---

This PR is mainly just checks (as is often the case with code) and then code to show the current configuration and code to let you call `saSetMode` with 0 (clear) or `SA_MODE_SET_IN_RANGE_PITMODE` or its out-range equivalent.

---

Completely aside from the above - the SA flags are very simple but it took a bit of fiddling around for me to understand how they're supposed to be used. I've summarized their behavior for myself (and anyone else who's interested) [here](https://gist.github.com/george-hawkins/9fbf557a41dcb063c480cd9152de6d32). `vtxCommonSetPitMode` covers the cases numbered 2, 4 and 5 there and the SmartAudio OSD submenu (and this PR) cover the cases numbered 1 and 3 (for both `SA_MODE_SET_IN_RANGE_PITMODE` and `SA_MODE_SET_OUT_RANGE_PITMODE`).

Reply:
Yes, I think the idea to integrate these configuration settings into the unified VTX stack (potentially as flags on the value supplied into `vtxCommonSetPitMode()`) is the right way to go. I do get your point that for devices requiring complex runtime configuration this isn't viable, but I don't think that the complexity of VTX devices will get to this level - the protocols, including configuration are fairly simple.
The CMS menus for VTX are a leftover from the time before there was a unified stack, so they should not be used as a template / justification to not unify this.